### DISCUSSION TITLE
Check whether the periodic chart is correct

### DISCUSSION THREAD

**Discussion by HomWang** [nuxt/nuxt#17052] [Questions]
![image](https://user-images.githubusercontent.com/17541676/185315588-13d34c36-84bb-4eb5-93fa-777b3911059e.png)

**Comment by danielroe** [comment]
Nice diagram! It might be a bit misleading as the order of the bottom half doesn't represent the order of how these files are loaded/run. What is the purpose/what is it intending to show?

**Reply by HomWang** [reply]
Help more prople
Link: https://github.com/nuxt/framework/issues/6719

**Reply by HomWang** [reply]
Check `server` and `client` console:
https://stackblitz.com/edit/github-jjxw1t-9bweby?file=pages%2FtestTsx1.tsx,pages%2FtestTsx2.tsx,middleware%2Fpublic.global.ts

**Reply by danielroe** [reply]
Great idea 👍 I think we should work on making this clearer, and a lifecycle diagram is a good idea. (@Krutie has made some amazing ones in the past!)

The difference you're seeing is between when a component/page is _imported_ (which might be dynamic) and when it is executed/rendered (which is also when the setup function is called, etc.).

In TSX files (and indeed any component with `<script>` that isn't `<script setup>`) anything that is outside the component definition will be run when the file is imported, not when the component is rendered.

**Reply by HomWang** [reply]
This helped me a lot, thanks
